ooni/nettests/experimental/tls_handshake.py

```diff
@@ -126,9 +129,11 @@ class HandshakeOptions(usage.Options):
          'Remote host IP address (v4/v6) and port, i.e. "1.2.3.4:443"'],
         ['port', 'p', None,
          'Use this port for all hosts, regardless of port specified in file'],
+        ['sni', None, None, 'Use SNI for Server Name Indication field'],
         ['ciphersuite', 'c', None ,
          'File containing ciphersuite list, one per line'],]
     optFlags = [
+        ['no-sni', None, 'Disable SNI'],
```

Comment:
I don't fully understand the purpose of this flag. It seems like the default behaviour is to **not** use SNI, so in which cases would this flag trigger a different behaviour.

I see you added a note below about this, perhaps we can either remove the flag or make the default behaviour be inferred depending on whether or not we have SNI support.